### ISSUE TITLE
Should markers ignore case when comparing packages?

### ISSUE THREAD

**Issue by jeroendecroos** [pypa/packaging#507]
In https://github.com/pypa/pip/issues/10889 I found that markers are not matched when requesting an extra that has uppercase (as is defined in the package)
e.g.:

wheel:

```
$ rg Mustache gitchangelog-3.0.4
gitchangelog-3.0.4/gitchangelog-3.0.4.dist-info/METADATA
26:Provides-Extra: Mustache
27:Requires-Dist: pystache; extra == 'Mustache'
```

Trying to install this with extras:
```
$ pip install gitchangelog[Mustache]
Collecting gitchangelog[mustache]
  Downloading https://files.pythonhosted.org/packages/9d/f9/554f9d1e2031a330148299b81ed7d6d5f1ef8969154384ca4ae8dd403b7a/gitchangelog-3.0.4-py2.py3-none-any.
whl (54kB)
     |████████████████████████████████| 61kB 7.9MB/s
  Ignoring pystache: markers 'extra == "Mustache"' don't match your environment
Installing collected packages: gitchangelog
Successfully installed gitchangelog-3.0.4
```

I've found a fix and will open up a PR

**Comment by pradyunsg** [comment]
Please don't mention issue numbers in commits. They result in spam on the issue, as above, and is generally not particularly useful since commits from merged PRs will hold a reference to the relevant PR in the commit page.

**Comment by jeroendecroos** [comment]
Issue was solved with the implementation of PEP 685 (#526)
